src/sharedHooks/useAuthenticatedQuery.js

```diff
@@ -0,0 +1,17 @@
+// @flow
+import { useQuery } from "@tanstack/react-query";
+
+import { getJWTToken } from "../components/LoginSignUp/AuthenticationService";
+
+const useAuthenticatedQuery = (
+  queryKey: string,
+  query: Function
+): any => useQuery( [queryKey], async ( ) => {
+  const apiToken = await getJWTToken( );
+  const options = {
+    api_token: apiToken
+  };
+  return query( options );
+} );
+
+export default useAuthenticatedQuery;
```

Comment:
Interesting, but I think the query results will get cached and survive sign-out, no? We might want to make sure every query key array includes the user ID. You could also `useState` to store the JWT until it stops working and do the error handling for that here.